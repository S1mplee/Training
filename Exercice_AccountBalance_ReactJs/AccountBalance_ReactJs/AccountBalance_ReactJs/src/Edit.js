import React, { Component } from 'react';
import dotnetify from 'dotnetify';

class Edit extends Component {
    constructor(props) {
        super();
        this.vm = dotnetify.react.connect("EditViewModel", this);
        this.state = {
            ID: "",
            over: "",
            daily: "",
            message:""
        };
        this.styles = {
            color: "green"
        };

        this.dispatchState = state => {
            this.setState(state);
            this.vm.$dispatch(state);
        };

        this.state.ID = props.value;
    }

    componentWillUnmount() {
        this.vm.$destroy();
    }
  

    render() {
        return (<div className='popup'>
            <div className='popup_inner'>
                <button className="btn btn-danger m-2" onClick={this.props.closePopup}>x</button>
<div class="card mx-xl-5">
                    <label style={{ color: 'red' }}>{this.state.message} </label>                     

  <div class="card-body">

                        <p class="h5 text-center py-4" style={{ color: 'green' }}>Edit Account: </p>

      <label class="grey-text font-weight-light text-center">OverDraft Limit :</label>
                        <input onBlur={_ => this.dispatchState({ over: this.state.over })}
                            onChange={e => this.setState({ over: e.target.value })} value={this.state.over} type="text" id="defaultFormCardNameEx" class="form-control"/>

                                <br/>

                        <label class="grey-text font-weight-light">Daily Wire Transfert </label>
                        <input onBlur={_ => this.dispatchState({ daily: this.state.daily })}
                            onChange={e => this.setState({ daily: e.target.value })} value={this.state.daily} type="email" id="defaultFormCardEmailEx" class="form-control"/>
                        <br />

                        <div class="text-center py-4 mt-3">
                            <button onClick={_ => this.dispatchState({ ButtonClicked: this.props.value })} class="btn btn-warning" type="">Send<i
                                                class="fa fa-paper-plane-o ml-2"></i></button>
                                        </div>
                                
                    </div>
</div>
            </div>
        </div> );
    }
}
 
export default Edit;