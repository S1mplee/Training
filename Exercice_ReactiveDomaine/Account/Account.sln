
Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio 15
VisualStudioVersion = 15.0.28307.645
MinimumVisualStudioVersion = 10.0.40219.1
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Account", "Account\Account.csproj", "{FE63F6F7-F42F-4553-B887-BA956FFDDB36}"
EndProject
Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Test", "Test\Test.csproj", "{67116D0A-DB50-4D1B-B3AD-7A2FAD936563}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{FE63F6F7-F42F-4553-B887-BA956FFDDB36}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{FE63F6F7-F42F-4553-B887-BA956FFDDB36}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{FE63F6F7-F42F-4553-B887-BA956FFDDB36}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{FE63F6F7-F42F-4553-B887-BA956FFDDB36}.Release|Any CPU.Build.0 = Release|Any CPU
		{67116D0A-DB50-4D1B-B3AD-7A2FAD936563}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{67116D0A-DB50-4D1B-B3AD-7A2FAD936563}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{67116D0A-DB50-4D1B-B3AD-7A2FAD936563}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{67116D0A-DB50-4D1B-B3AD-7A2FAD936563}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(ExtensibilityGlobals) = postSolution
		SolutionGuid = {E51E8A11-6A11-444F-81B2-C881CABEEECC}
	EndGlobalSection
EndGlobal
