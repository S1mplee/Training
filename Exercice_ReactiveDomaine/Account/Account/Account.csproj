<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <Import Project="..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.visualstudio.2.4.0\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.visualstudio.2.4.0\build\net20\xunit.runner.visualstudio.props')" />
  <Import Project="..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.console.2.4.0\build\xunit.runner.console.props" Condition="Exists('..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.console.2.4.0\build\xunit.runner.console.props')" />
  <Import Project="..\packages\xunit.core.2.4.1\build\xunit.core.props" Condition="Exists('..\packages\xunit.core.2.4.1\build\xunit.core.props')" />
  <Import Project="..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.props" Condition="Exists('..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.props')" />
  <Import Project="..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.props" Condition="Exists('..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.props')" />
  <Import Project="..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props" Condition="Exists('..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props')" />
  <Import Project="..\packages\xunit.runner.console.2.3.1\build\xunit.runner.console.props" Condition="Exists('..\packages\xunit.runner.console.2.3.1\build\xunit.runner.console.props')" />
  <Import Project="$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props" Condition="Exists('$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\Microsoft.Common.props')" />
  <PropertyGroup>
    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
    <ProjectGuid>{FE63F6F7-F42F-4553-B887-BA956FFDDB36}</ProjectGuid>
    <OutputType>Exe</OutputType>
    <RootNamespace>Account</RootNamespace>
    <AssemblyName>Account</AssemblyName>
    <TargetFrameworkVersion>v4.6.1</TargetFrameworkVersion>
    <FileAlignment>512</FileAlignment>
    <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
    <Deterministic>true</Deterministic>
    <NuGetPackageImportStamp>
    </NuGetPackageImportStamp>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Debug|AnyCPU' ">
    <PlatformTarget>AnyCPU</PlatformTarget>
    <DebugSymbols>true</DebugSymbols>
    <DebugType>full</DebugType>
    <Optimize>false</Optimize>
    <OutputPath>bin\Debug\</OutputPath>
    <DefineConstants>DEBUG;TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <PropertyGroup Condition=" '$(Configuration)|$(Platform)' == 'Release|AnyCPU' ">
    <PlatformTarget>AnyCPU</PlatformTarget>
    <DebugType>pdbonly</DebugType>
    <Optimize>true</Optimize>
    <OutputPath>bin\Release\</OutputPath>
    <DefineConstants>TRACE</DefineConstants>
    <ErrorReport>prompt</ErrorReport>
    <WarningLevel>4</WarningLevel>
  </PropertyGroup>
  <ItemGroup>
    <Reference Include="EventStore.ClientAPI, Version=0.0.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\EventStore.Client.5.0.1\lib\net46\EventStore.ClientAPI.dll</HintPath>
    </Reference>
    <Reference Include="Microsoft.VisualStudio.CodeCoverage.Shim, Version=15.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
      <HintPath>..\packages\Microsoft.CodeCoverage.16.2.0\lib\net45\Microsoft.VisualStudio.CodeCoverage.Shim.dll</HintPath>
    </Reference>
    <Reference Include="Newtonsoft.Json, Version=12.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
      <HintPath>..\..\..\..\..\source\repos\Reactjs_Account\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="Newtonsoft.Json.Bson, Version=1.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
      <HintPath>..\packages\Newtonsoft.Json.Bson.1.0.1\lib\net45\Newtonsoft.Json.Bson.dll</HintPath>
    </Reference>
    <Reference Include="PresentationFramework" />
    <Reference Include="protobuf-net, Version=2.4.0.0, Culture=neutral, PublicKeyToken=257b51d87d2e4d67, processorArchitecture=MSIL">
      <HintPath>..\packages\protobuf-net.2.4.0\lib\net40\protobuf-net.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Buffers, Version=0.8.18.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\..\..\..\..\source\repos\Reactjs_Account\packages\ReactiveDomain.Buffers.0.8.18\lib\net452\ReactiveDomain.Buffers.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Core, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Core.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Foundation, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Foundation.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Messaging, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Messaging.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Persistence, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Persistence.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.PrivateLedger, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.PrivateLedger.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Testing, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Testing.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.Transport, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.Transport.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.UI, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.UI.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveDomain.UI.Testing, Version=0.8.21.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\ReactiveDomain.0.8.21\lib\net452\ReactiveDomain.UI.Testing.dll</HintPath>
    </Reference>
    <Reference Include="ReactiveUI, Version=7.4.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\reactiveui-core.7.4.0\lib\Net45\ReactiveUI.dll</HintPath>
    </Reference>
    <Reference Include="Splat, Version=1.4.0.0, Culture=neutral, processorArchitecture=MSIL">
      <HintPath>..\packages\Splat.1.4.0\lib\Net45\Splat.dll</HintPath>
    </Reference>
    <Reference Include="System" />
    <Reference Include="System.Configuration" />
    <Reference Include="System.Core" />
    <Reference Include="System.Reactive.Core, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <HintPath>..\packages\Rx-Core.2.2.5\lib\net45\System.Reactive.Core.dll</HintPath>
    </Reference>
    <Reference Include="System.Reactive.Interfaces, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <HintPath>..\packages\Rx-Interfaces.2.2.5\lib\net45\System.Reactive.Interfaces.dll</HintPath>
    </Reference>
    <Reference Include="System.Reactive.Linq, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <HintPath>..\packages\Rx-Linq.2.2.5\lib\net45\System.Reactive.Linq.dll</HintPath>
    </Reference>
    <Reference Include="System.Reactive.PlatformServices, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <HintPath>..\packages\Rx-PlatformServices.2.2.5\lib\net45\System.Reactive.PlatformServices.dll</HintPath>
    </Reference>
    <Reference Include="System.Reactive.Windows.Threading, Version=2.2.5.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
      <HintPath>..\packages\Rx-XAML.2.2.5\lib\net45\System.Reactive.Windows.Threading.dll</HintPath>
    </Reference>
    <Reference Include="System.Runtime.CompilerServices.Unsafe, Version=4.0.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL">
      <HintPath>..\..\..\..\..\source\repos\Reactjs_Account\packages\System.Runtime.CompilerServices.Unsafe.4.5.1\lib\netstandard2.0\System.Runtime.CompilerServices.Unsafe.dll</HintPath>
    </Reference>
    <Reference Include="System.Runtime.Serialization" />
    <Reference Include="System.ServiceModel" />
    <Reference Include="System.Xml.Linq" />
    <Reference Include="System.Data.DataSetExtensions" />
    <Reference Include="Microsoft.CSharp" />
    <Reference Include="System.Data" />
    <Reference Include="System.Net.Http" />
    <Reference Include="System.Xml" />
    <Reference Include="WindowsBase" />
    <Reference Include="xunit.abstractions, Version=2.0.0.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
      <HintPath>..\packages\xunit.abstractions.2.0.3\lib\net35\xunit.abstractions.dll</HintPath>
      <Private>True</Private>
    </Reference>
    <Reference Include="xunit.assert, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
      <HintPath>..\packages\xunit.assert.2.4.1\lib\netstandard1.1\xunit.assert.dll</HintPath>
    </Reference>
    <Reference Include="xunit.core, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
      <HintPath>..\packages\xunit.extensibility.core.2.4.1\lib\net452\xunit.core.dll</HintPath>
    </Reference>
    <Reference Include="xunit.execution.desktop, Version=2.4.1.0, Culture=neutral, PublicKeyToken=8d05b1bb7a6fdb6c, processorArchitecture=MSIL">
      <HintPath>..\packages\xunit.extensibility.execution.2.4.1\lib\net452\xunit.execution.desktop.dll</HintPath>
    </Reference>
  </ItemGroup>
  <ItemGroup>
    <Compile Include="account.cs" />
    <Compile Include="AccountBalance.cs" />
    <Compile Include="CommandHandler.cs" />
    <Compile Include="Commands\CreateAccount.cs" />
    <Compile Include="Commands\DeposeCash.cs" />
    <Compile Include="Commands\DeposeCheque.cs" />
    <Compile Include="Commands\TransferCash.cs" />
    <Compile Include="Commands\WithDrawCash.cs" />
    <Compile Include="Events\AccountBlocked.cs" />
    <Compile Include="Events\AccountCreatedd.cs" />
    <Compile Include="Events\AccountUnblocked.cs" />
    <Compile Include="Events\CashDeposed.cs" />
    <Compile Include="Events\CashTransfered.cs" />
    <Compile Include="Events\CashWithdrawn.cs" />
    <Compile Include="Events\ChequeDeposed.cs" />
    <Compile Include="Program.cs" />
    <Compile Include="Properties\AssemblyInfo.cs" />
    <Compile Include="ReadModel.cs" />
    <Compile Include="service.cs" />
  </ItemGroup>
  <ItemGroup>
    <None Include="App.config" />
    <None Include="packages.config" />
  </ItemGroup>
  <ItemGroup>
    <Analyzer Include="..\packages\xunit.analyzers.0.10.0\analyzers\dotnet\cs\xunit.analyzers.dll" />
  </ItemGroup>
  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
  <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
    <PropertyGroup>
      <ErrorText>This project references NuGet package(s) that are missing on this computer. Use NuGet Package Restore to download them.  For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
    </PropertyGroup>
    <Error Condition="!Exists('..\packages\xunit.runner.console.2.3.1\build\xunit.runner.console.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\xunit.runner.console.2.3.1\build\xunit.runner.console.props'))" />
    <Error Condition="!Exists('..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\xunit.runner.visualstudio.2.3.1\build\net20\xunit.runner.visualstudio.props'))" />
    <Error Condition="!Exists('..\packages\EventStore.Client.Embedded.5.0.1\build\EventStore.Client.Embedded.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\EventStore.Client.Embedded.5.0.1\build\EventStore.Client.Embedded.targets'))" />
    <Error Condition="!Exists('..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.props'))" />
    <Error Condition="!Exists('..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.targets'))" />
    <Error Condition="!Exists('..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.props'))" />
    <Error Condition="!Exists('..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.targets'))" />
    <Error Condition="!Exists('..\packages\xunit.core.2.4.1\build\xunit.core.props')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\xunit.core.2.4.1\build\xunit.core.props'))" />
    <Error Condition="!Exists('..\packages\xunit.core.2.4.1\build\xunit.core.targets')" Text="$([System.String]::Format('$(ErrorText)', '..\packages\xunit.core.2.4.1\build\xunit.core.targets'))" />
    <Error Condition="!Exists('..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.console.2.4.0\build\xunit.runner.console.props')" Text="$([System.String]::Format('$(ErrorText)', '..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.console.2.4.0\build\xunit.runner.console.props'))" />
    <Error Condition="!Exists('..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.visualstudio.2.4.0\build\net20\xunit.runner.visualstudio.props')" Text="$([System.String]::Format('$(ErrorText)', '..\..\..\..\..\source\repos\Reactjs_Account\packages\xunit.runner.visualstudio.2.4.0\build\net20\xunit.runner.visualstudio.props'))" />
  </Target>
  <Import Project="..\packages\EventStore.Client.Embedded.5.0.1\build\EventStore.Client.Embedded.targets" Condition="Exists('..\packages\EventStore.Client.Embedded.5.0.1\build\EventStore.Client.Embedded.targets')" />
  <Import Project="..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.targets" Condition="Exists('..\packages\Microsoft.CodeCoverage.16.2.0\build\netstandard1.0\Microsoft.CodeCoverage.targets')" />
  <Import Project="..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.targets" Condition="Exists('..\packages\Microsoft.NET.Test.Sdk.16.2.0\build\net40\Microsoft.NET.Test.Sdk.targets')" />
  <Import Project="..\packages\xunit.core.2.4.1\build\xunit.core.targets" Condition="Exists('..\packages\xunit.core.2.4.1\build\xunit.core.targets')" />
</Project>