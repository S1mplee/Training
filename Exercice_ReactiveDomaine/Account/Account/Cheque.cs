using System;

namespace TestAccountBalance
{
    public class Cheque
    {
        public decimal Amount;
        public DateTime Date;
        public bool Checked;

        public Cheque(decimal amount,DateTime date)
        {
            this.Amount = amount;
            this.Date = date;
            this.Checked = false;
        }
    }
}