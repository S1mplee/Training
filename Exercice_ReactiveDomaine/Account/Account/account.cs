using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Account
{
    public class account
    {
        public Guid Id { get; set; }
        public string HolderName { get; set; }
        public decimal cash { get; set; }
        public decimal overdraftlimit { get; set; }
        public decimal wiretransfertlimit { get; set; }
        public bool blocked { get; set; }
    }
}
