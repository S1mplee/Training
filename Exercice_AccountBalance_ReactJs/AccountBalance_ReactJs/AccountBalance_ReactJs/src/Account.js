import React, { Component } from 'react';
import dotnetify from 'dotnetify';
import Edit from './Edit';

class Account extends Component {
    constructor()
    {
        super();
        this.vm = dotnetify.react.connect("AccountViewModel", this);

        this.state = {
            counter: 0,
            tags: [{ name: "Ali", Age: "18" }, { name: "Mohamed", Age: "20" }, { name: "Fathi", Age: "20" }],
            age: 0,
            list: [],
            value: "",
            g1: "",
            g2: "",
            message: "",
            showPopup: false,
            ID:""
        };
        this.handleme = this.handleme.bind(this);
        this.styles = {
            color: "green"
        };

        this.dispatchState = state => {
            this.setState(state);
            this.vm.$dispatch(state);
        };
    }

    componentWillUnmount() {
        this.vm.$destroy();
    }

   
    
    ErrorMessage() {
        if (this.state.message !== "") {
            return <div class="alert alert-danger" role="alert">
                <p className="text-center"> {this.state.message} </p>
            </div>
        }
    }

    handleme () {
        window.open(Navbar, '_blank');
    };

    

    render() { 
        const togglePopup = (key) => {
            this.setState({
                showPopup: !this.state.showPopup
            });

            this.dispatchState({ ID: key });
        };

        const Status = (bool) => {
            if (bool) return "table-danger";
            return "table-success";
        };

        return (<div className="container">
            {this.ErrorMessage()}
            <div className="table-wrapper">
                <div className="table-title">
                    <div className="row">
                    <div class="col-sm-6">
						<h2>Manage <b>AccountBalance</b></h2>
					</div>
                        <div class="col-sm-6"  >
                            <div class="form-group col-sm-4 myclass">
                                <label for="focusedInput">Holder Name</label>
                                <input class="form-control" onBlur={_ => this.dispatchState({ name: this.state.name })}
                                    onChange={e => this.setState({ name: e.target.value })} value={this.state.name} class="form-control" id="focusedInput" type="text" />
                            </div>
                            
                          
                            <a href="/add/" onClick={_ => this.dispatchState({ clicked: true })} class="btn btn-success" data-toggle="modal"><i class="material-icons">&#xE147;</i> <span>Add New Account</span></a>
    
                   

					</div>
                </div>
            </div>
            <table class="table table-striped table-hover">
                <thead>
                    <tr>
                            <th>
                                Day
						</th>
                        <th>NameHolder</th>
                        <th>Balance</th>
						<th>Over draft Limit</th>
                        <th>Wire Transfert limit</th>
                        <th>Actions</th>
                    </tr>
                </thead>
                    <tbody>
                        {this.state.list.map(acc => <tr className={Status(acc.blocked)}>
						<td>
                                <span>
                                    {acc.DailyWireTransfetAchieved}
							</span>
                            </td>
                            <td class="table-success" ><p>{acc.HolderName}</p></td>
                            <td>{acc.cash}</td>
                            <td>{acc.overdraftlimit}</td>
                            <td>{acc.wiretransfertlimit}</td>
                            <td>
                                <div className="container">
                                    <div className="row">
                                        <div className="col-xs-4">
                                            <button disabled={acc.blocked} onClick={a => this.dispatchState({ accountid: acc.Id })} className="btn btn-danger btn-sm m-1">WithDraw</button>
                                        </div>
                                        <div className="col-xs-4 text-center">
                                            <button className="btn btn-success  btn-sm m-1" onClick={e => this.dispatchState({ id: acc.Id })}>Depose</button>
                                        </div>
                                        <div className="col-xs-2 text-right m-1">
                                            <div class="input-group mb-3">
                                                <div class="input-group-prepend">
                                                    <span class="input-group-text">$</span>
                                                    <span class="input-group-text"></span>
                                                </div>
                                                <input style={{width : '70px'}} type="text" class="form-control" onBlur={_ => this.dispatchState({ value: this.state.value })}
                                                    onChange={e => this.setState({ value: e.target.value })} value={this.state.value} aria-label="Amount (to the nearest dollar)" />
                                                <button className="btn btn-info  btn-sm m-2" onClick={(e) => togglePopup(acc.Id, e)}>Edit </button>

                                                <button disabled={acc.blocked} onClick={e => this.dispatchState({ Tid: acc.Id })}  className="btn btn-warning  btn-sm m-2">Transf </button>
                                                
                                            </div>                                            
                                          
                                        </div>
                                    </div>
                                </div>
                            </td>
                    </tr> )}
                   
                        {this.state.showPopup ?
                            <Edit
                                text='Close Me'
                                closePopup={togglePopup}
                                value={this.state.ID}
                            />
                            : null
                        }
                </tbody>
            </table>
            </div>
           
           
    </div>
	  );
    }
   
}
 
export default Account;