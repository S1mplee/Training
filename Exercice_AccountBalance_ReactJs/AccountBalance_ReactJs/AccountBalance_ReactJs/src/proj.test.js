import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import Edit from './Edit';
import Account from './Account';

test(" Ui snapshot Render for Account", () => {
    const tree = renderer
        .create(<Account />)
        .toJSON();
    expect(tree).toMatchSnapshot();

});

test(" Ui snapshot Render for Edit", () => {
    const tree = renderer
        .create(<Edit />)
        .toJSON();
    expect(tree).toMatchSnapshot();

});

test('Account : Testing Container Render', () => {
    const component = shallow(<Account />);
    const wrapper = component.find('.container');
    expect(wrapper.length).toBeGreaterThan(0);

});

test('Edit : Testing popup Render', () => {
    const component = shallow(<Edit />);
    const wrapper = component.find('.popup');
    expect(wrapper.length).toBeGreaterThan(0);

});

test('Status Fonction : Testing true return ', () => {
    var res = Status(true);

    expect(res).toBe("table-danger");

});

test('Status Fonction : Testing false return ', () => {
    var res = Status(false);

    expect(res).toBe("table-success");

});

test('ErrorMessage Fonction : Testing empty string ', () => {
    var res = ErrorMessage("");

    expect(res).toBe(undefined);

});

test('ErrorMessage Fonction : Testing empty string ', () => {

    var node = <div class="alert alert-danger" role="alert">
        < p className="text-center" >
            Hello
            </p >
        </div >
    var res = ErrorMessage("Hello");

    expect(res).toEqual(node);

});

test('TotalValue : Testing Total prop Render', () => {

    const tree = renderer
        .create(<Edit value="sqd546-qsdq56-dsq89" />)
    var instance = tree.getInstance();
    expect(instance.props.value).toBe("sqd546-qsdq56-dsq89");

});

const Status = (bool) => {
    if (bool) return "table-danger";
    return "table-success";
};

const ErrorMessage = (message) => {
    if (message !== "") {
        return <div class="alert alert-danger" role="alert">
            <p className="text-center">{message}</p>
        </div>
    }
}


