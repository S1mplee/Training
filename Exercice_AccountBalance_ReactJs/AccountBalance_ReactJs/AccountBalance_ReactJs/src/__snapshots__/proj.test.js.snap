// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` Ui snapshot Render for Account 1`] = `
<div
  className="container"
>
  <div
    className="table-wrapper"
  >
    <div
      className="table-title"
    >
      <div
        className="row"
      >
        <div
          class="col-sm-6"
        >
          <h2>
            Manage 
            <b>
              AccountBalance
            </b>
          </h2>
        </div>
        <div
          class="col-sm-6"
        >
          <div
            class="form-group col-sm-4 myclass"
          >
            <label
              for="focusedInput"
            >
              Holder Name
            </label>
            <input
              class="form-control"
              id="focusedInput"
              onBlur={[Function]}
              onChange={[Function]}
              type="text"
            />
          </div>
          <a
            class="btn btn-success"
            data-toggle="modal"
            href="/add/"
            onClick={[Function]}
          >
            <i
              class="material-icons"
            >
              î…‡
            </i>
             
            <span>
              Add New Account
            </span>
          </a>
        </div>
      </div>
    </div>
    <table
      class="table table-striped table-hover"
    >
      <thead>
        <tr>
          <th>
            Day
          </th>
          <th>
            NameHolder
          </th>
          <th>
            Balance
          </th>
          <th>
            Over draft Limit
          </th>
          <th>
            Wire Transfert limit
          </th>
          <th>
            Actions
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[` Ui snapshot Render for Edit 1`] = `
<div
  className="popup"
>
  <div
    className="popup_inner"
  >
    <button
      className="btn btn-danger m-2"
    >
      x
    </button>
    <div
      class="card mx-xl-5"
    >
      <label
        style={
          Object {
            "color": "red",
          }
        }
      >
        
         
      </label>
      <div
        class="card-body"
      >
        <p
          class="h5 text-center py-4"
          style={
            Object {
              "color": "green",
            }
          }
        >
          Edit Account: 
        </p>
        <label
          class="grey-text font-weight-light text-center"
        >
          OverDraft Limit :
        </label>
        <input
          class="form-control"
          id="defaultFormCardNameEx"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value=""
        />
        <br />
        <label
          class="grey-text font-weight-light"
        >
          Daily Wire Transfert 
        </label>
        <input
          class="form-control"
          id="defaultFormCardEmailEx"
          onBlur={[Function]}
          onChange={[Function]}
          type="email"
          value=""
        />
        <br />
        <div
          class="text-center py-4 mt-3"
        >
          <button
            class="btn btn-warning"
            onClick={[Function]}
            type=""
          >
            Send
            <i
              class="fa fa-paper-plane-o ml-2"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
